devops/kubernetes/node-network/index.ts

```diff
@@ -1,342 +1,339 @@
-import * as awsx from '@pulumi/awsx'
-import * as eks from '@pulumi/eks'
 import * as pulumi from '@pulumi/pulumi'
 import * as k8s from '@pulumi/kubernetes'
 import { configMapFromFile } from './configMap'
-import { CaddyServiceDeployment } from 'pulumi-common'
+import { CaddyServiceDeployment, getProvider, isPlatformMinikube } from 'pulumi-common'
 import { getSubkeyContainers } from './utils'
 import { ValidatorServiceDeployment } from './validator'
 import { NFSServiceDeployment } from './nfsVolume'
-// const { exec } = require('child_process')
+import { readFileSync } from 'fs'
 
-const config = new pulumi.Config()
-const awsConfig = new pulumi.Config('aws')
-const isMinikube = config.getBoolean('isMinikube')
+export = async () => {
+  const config = new pulumi.Config()
 
-export let kubeconfig: pulumi.Output<any>
+  const kubeconfigFile = config.get('kubeconfigFile')
+  const clusterStackRefInput = config.get('clusterStackRef')
+  let provider: k8s.Provider = new k8s.Provider('local', {})
+  let isMinikube: boolean = false
```

Comment:
I think we should drop the minikube config but replace it with `useLocalProvider` for user to have freedom to just use the currently selected context in their default kubeconfig?